src/components/AutoUpdateTime.js

```diff
@@ -0,0 +1,86 @@
+import {View} from 'react-native';
+import React, {PureComponent} from 'react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import DateUtils from '../libs/DateUtils';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import Timers from '../libs/Timers';
+import Text from './Text';
+
+const propTypes = {
+    timezone: PropTypes.shape({
+    // Value of selected timezone
+        selected: PropTypes.string,
+
+        // Whether timezone is automatically set
+        automatic: PropTypes.bool,
+    }),
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    timezone: null,
+};
+
+class AutoUpdateTime extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
+        this.state = {
+            timezone: this.getCurrentUserLocalTime(),
+        };
+    }
+
+    componentDidMount() {
+        this.timer = Timers.register(setInterval(() => {
+            this.setState({
+                timezone: this.getCurrentUserLocalTime(),
+            });
+        }, 60000));
```

Comment:
This isn't right, because we don't want to update the time in the middle of the minute (eg, at 15:31:27), we want to update it as close to the top of the minute as possible, eg, 15:31:00 or 15:31:01, so this needs to be queued for that time

Reply:
@cead22 yeah, this make senses.
But can you explain more about this? We should make the value `60000` dynamically based on the last minutes we updated right?

Reply:
Yeah, if we're 36 seconds away from the next minute, we schedule for 36s in the future. After that we can use 60s or simply recalculate this which is probably easier since we have to make the calculation the first time

Reply:
@cead22 Finally I did it 😂 , please checkout my latest code. It took me a while to do that.